e2e_example/tool/build.dart

```diff
@@ -1,12 +1,66 @@
-// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+
 import 'dart:async';
+import 'dart:io';
 
+import 'package:build/build.dart';
+import 'package:build_compilers/build_compilers.dart';
 import 'package:build_runner/build_runner.dart';
-
-import 'phases.dart';
+import 'package:build_test/builder.dart';
+import 'package:shelf/shelf_io.dart' as shelf_io;
 
 Future main() async {
-  await build(phases);
+  var graph = new PackageGraph.forThisPackage();
+  var buildActions = <BuildAction>[
+    new BuildAction(new TestBootstrapBuilder(), graph.root.name,
+        inputs: ['test/**_test.dart']),
+  ];
+
+  void crawlPackageAndAddBuilder(
```

Comment:
I think it would be nice to use the `orderedPackages` and `dependentsOf` from the `PackageGraph` rather than a custom function here - I think this is what the generated scripts will end up using.

See for example: https://gist.github.com/natebosch/3f17d779a068dcf04f0b0ccbcb00edb9#file-serve-dart for using `dependentsOf` with angular.

Reply:
done, using `orderedPackages`